### ISSUE TITLE
[A11y]: Onramping Tool - Information Architecture - Design Approach Input

### ISSUE THREAD

**Issue by kbermangestring** [department-of-veterans-affairs/va.gov-team#115559]
# 🔧 Accessibility Support Request

## I am requesting...

**Design / Research**

- [X] Design [approach] review
- [ ] A11y annotations
- [ ] Prototype testing
- [ ] User research planning
- [ ] Tech support for research sessions

**Dev / QA / Training**

- [ ] A11y auditing & testing
- [ ] Collaboration Cycle accessibility testing artifact
- [ ] PR Review - *No need to fill this ticket out, just tag me as a reviewer*
- [ ] A11y training
- [ ] Office Hours / Deep Dive

## Background
- **Team name**:Decision Review
- **Product name**: Onramping Tool
- **Feature name**: n/a
- **Contact person**: @kbermangestring 
- **Next Collaboration Cycle touchpoint & date if applicable:**
- **Link to PR/Figma/prototype etc**:https://www.figma.com/design/5vAWK3wpBkJgG7ngLXYmht/Onramping-Tool?node-id=290-13429&t=7TklreolE5W0wHGE-11
- **Links to any related GitHub  #issues**:  [#113548](https://github.com/department-of-veterans-affairs/va.gov-team/issues/113548) 
- **Additional details on your request**:

- Hey, would love your initial thoughts on exploring different component use/ways of organizing information for the Onramping tool. We are hoping to provide context to users who may need more information or resources at each step in the onramping tool questionnaire flow, but want to be mindful of making it as seamless an experience (with attention to AT users) as possible. 

We have 4 explorations for feedback:
A- Tool title, radio button single form component variation (wondering if this is too much space between question and radio selections & if there is a verbosity concern with having the "onramping tool" title on every screen?
B- no tool title, general title, content, radio button
C- no tool title, question title, content, radio button (this duplicates the question in the title and in the radio button component)
D- no tool title, general title, content, radio button with h3 

Would love a quick, high-level input on what to consider for accessibility as we work on flushing out the design. 

## Yes, I have

- [ ] Formatted the GitHub issue title to include: [A11y]: Product - Feature - Request
- [ ] Set my project under "Projects"
- [ ] Added the appropriate team/project label under "Labels" (e.g., Decision Reviews, BMT1, or BMT2)
- [ ] Verified that @bacitracin is assigned to this ticket

**Comment by bacitracin** [comment]
@kbermangestring  @conchette @cinlackey 
Okayyyy 🤓  I had a chance to review and here are my notes:

## Recommendation

### Use C with modifications
- I’d suggest sticking closely to the [Forms Pattern Single radio button (FPSRB) ](https://design.va.gov/components/form/radio-button#forms-pattern-single)if you can, there is built in labelling that connects the radio buttons with the header text & the description text. It’ll announce better for screen reader users without having to pass in extra text in a hidden area that may be repetitive for AT users. Even though wordy, this is essential information that we don’t want AT users to miss. So for AT users navigating in ["forms mode"](https://human-centred.nz/2022/02/10/the-two-modes-of-screen-readers/) they will hear all the text when they land on the radio button. See image of my light mock up using Storybook (missing the PACT act link):

<img width="1218" height="677" alt="Image" src="https://github.com/user-attachments/assets/d9274a81-2406-4d3b-93aa-ebfb42497b1e" />

- That said, the FPSRB announces its header reliably, but not the radio button group legend (the *required line) right above the radio buttons. So you will want the h1 to be the question “Are you requesting a review because of a change in law or policy?” and not “Change in law or policy”. You can see in my VoiceOver screenshot that “Select one *required” is not announced at all. That text is still available to AT users in "browse mode."

- Summary of C & any changes:
    - Title is h1 and is question
    - Use the FPSRB here instead of the regular radio button variation
    - Page h1 and the legend text matching may be confusing for AT users. Ideally change the text of the legend from repeating the heading to something else, but not a dealbreaker.
    - You’ll want to doublecheck that you can use the radio button tile with FPSRB. The tile variation is new to me, so maybe it works, maybe it doesn’t not sure.

### Next best: D
- If you run into problems implementing FPSRB or decide not to go that route for whatever reason, then option D is the next best (and very similar to what we’ve got going on on the Add conditions page of the 526 disability form).
    - If we’re not using the FPSRB, then we’re foregoing all the built in announcements and accepting that screen reader users TAB-bing around may miss our explanatory text and bullets. We can pass in hidden messages to be announced when landing on the radio button group, but as I mentioned earlier I consider it less ideal because it creates a duplicate hidden area with the text, and the text is more likely to get cutoff in announcement.
    - Having “Change in Law or Policy” as an H1 would match the breadcrumbs + url, so that’s actually a huge plus.
    - Having the “Are you requesting a review” legend as a header increases scannability. But as a general rule we can’t skip heading levels, we have to go in order. So it should be an H2 styled as an H3.
    - You’ll want to doublecheck that you can actually set the heading level to H2 for the radio button tile. I tried on Storybook but it wasn’t updating, which was weird, so just flagging. https://design.va.gov/storybook/?path=/story/uswds-va-radio--tile

## Other Notes
- Each page MUST have an H1, and we need to make each H1/breadcrumb/title unique on every page of this form flow. We have gotten dinged by the 508 office on this before. So just a heads up that we should avoid having “Onramp to decision review tool” as the same h1 on every page of this form. If you wanted to have something like "Onramp to decision review tool - Change in Policy" that could work. See https://github.com/department-of-veterans-affairs/va.gov-team/issues/105836
- Note about version A - I don’t know that this design is possible in the Forms Pattern Single Radio button pattern. The header can’t have *(*required).* That would go on the legend, which is missing from this design.
- This is kind of a touch and go bug with the tile version of the DS radio buttons. Sometimes each item counts as 4 items, so for the yes & no when landing on the radio button it would read out as “1 of 8” which would be super confusing to AT users. Each item is also read out 4 times on select screenreaders. See screenshot. I had this bug at first and then it just disappeared, so I’m not saying don’t use the tiled radio buttons, but watch out for this when we are testing our prototype. We may have to go back to regular radio buttons and make the description text part of the button label.

<img width="982" height="614" alt="Image" src="https://github.com/user-attachments/assets/343b05b5-4a2a-49f7-afea-cc37cc74554d" />

**Comment by bacitracin** [comment]
@randimays This is the ticket I mentioned. Outstanding design system things that I wasn't sure about:
1. _You’ll want to doublecheck that you can use the radio button tile with FPSRB. The tile variation is new to me, so maybe it works, maybe it doesn’t not sure._
2. _You’ll want to doublecheck that you can actually set the heading level to H2 for the radio button tile. I tried on Storybook but it wasn’t updating, which was weird, so just flagging. https://design.va.gov/storybook/?path=/story/uswds-va-radio--tile_

**Comment by bacitracin** [comment]
@randimays Thank you for looking into this! Sorry for clarification on question 2 - can https://design.va.gov/storybook/?path=/story/uswds-va-radio--label-header support the tile variation?

**Comment by randimays** [comment]
@bacitracin Yes, the Label Header version can support the tile variation. I applied it to the current dev build of the onramp, which is NOT FPSRB:

<img width="322" height="564" alt="Image" src="https://github.com/user-attachments/assets/260a64d2-78c8-4c0e-a793-52dee8ac11c6" />

Here's the DOM for reference:

```
<va-radio class="vads-u-margin-top--0 hydrated" data-testid="Q_1_2_CLAIM_DECISION" id="onramp-radio" label="Was your claim decided less than a year ago?" label-header-level="2" required="true" aria-invalid="false">
  <va-radio-option data-testid="va-radio-option" description="My claim was decided less than a year ago" label="Yes" name="group" tile="true" value="Yes" class="hydrated">
    <div class="usa-radio va-radio-option__container--tile">
      <input class="va-radio-option__input" type="radio" name="group" value="Yes" id="groupYesinput">
      <label class="usa-radio__label" for="groupYesinput">Yes
        <span class="usa-radio__label-description dd-privacy-hidden" data-dd-action-name="description">
          My claim was decided less than a year ago
        </span>
      </label>
    </div>
  </va-radio-option>
  <va-radio-option data-testid="va-radio-option" description="My claim was decided over a year ago" label="No" name="group" tile="true" value="No" class="hydrated">
    <div class="usa-radio va-radio-option__container--tile">
      <input class="va-radio-option__input" type="radio" name="group" value="No" id="groupNoinput">
      <label class="usa-radio__label" for="groupNoinput">No
        <span class="usa-radio__label-description dd-privacy-hidden" data-dd-action-name="description">
          My claim was decided over a year ago
        </span>
      </label>
    </div>
  </va-radio-option>
</va-radio>
```
